### PR TITLE
feat: add functionality to get values from redis commands

### PR DESCRIPTION
**Pull request by yquansah**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

This PR is an attempt to pull additional metrics on redis from arbitrary commands. The logic in this PR accounts for types (`string`, `integer`, and `float`), and we could add more types.

Issue addressed: https://github.com/influxdata/idpe/issues/8711
Community issue Addressed: https://github.com/influxdata/telegraf/issues/1027

### CODE REVIEW

**Comment by yquansah** [comment]
Was a little unsure of how exactly to go about editing README

File: plugins/inputs/redis/redis.go [hunk 1]
```diff
@@ -38,6 +46,37 @@ type RedisClient struct {
...
+func (r *RedisClient) Do(returnType string, args ...interface{}) (interface{}, error) {
+	rawVal := r.client.Do(args...)
+
+	switch returnType {
+	case "integer":
+		intVal, err := rawVal.Int64()
+		if err != nil {
+			return nil, err
+		}
+		return intVal, nil
```
Comment (ssoroka) [inline plugins/inputs/redis/redis.go:58]:
```suggestion
		return rawVal.Int64()
```

File: plugins/inputs/redis/redis.go [hunk 2]
```diff
@@ -64,6 +103,11 @@ var sampleConfig = `
   ## If no port is specified, 6379 is used
   servers = ["tcp://localhost:6379"]
 
+  ## Optional. Specify redis commands to retrieve values, under key "commands"
+  # command = ["get", "sample-key"]
+  # key = "sample-key-value"
+  # type = "string"
+
```
Comment (ssoroka) [inline plugins/inputs/redis/redis.go:110]: this looks wrong. Don't you need to specify `[[inputs.redis.commands]]` ahead of this?

Comment (ssoroka) [reply plugins/inputs/redis/redis.go:110]: consider naming "key" `field` instead.

Comment (ssoroka) [reply plugins/inputs/redis/redis.go:110]: changes should be copied into readme.

**Review by ssoroka — CHANGES_REQUESTED** [review]
Looks good. The only thing I'd say is that this might be better off as a new plugin. Already you're going to have a hard time changing the metric name if you don't like `redis_commands`. Also not clear how to control which commands go into which metric, which I guess is not possible currently. That said, I'm not necessarily against merging it into the existing plugin.

**Comment by reimda** [comment]
> Was a little unsure of how exactly to go about editing README

Describe how to use the feature you added.  Give examples so people unfamiliar with the feature will be able to understand how to configure it and know what data it will produce.

File: plugins/inputs/redis/redis.go (hunk 2 above)
Comment (yquansah) [reply plugins/inputs/redis/redis.go]: This is addressed! Thank you for the catch, let me know if there is anything else needed.

File: plugins/inputs/redis/redis_test.go [hunk 3]
```diff
@@ -7,11 +7,27 @@ import (
...
 	"github.com/influxdata/telegraf/testutil"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+type testClient struct {
+}
+
+func (t *testClient) BaseTags() map[string]string {
+	return map[string]string{"host": "redis.net"}
```
Comment (ssoroka) [inline plugins/inputs/redis/redis_test.go:20]: why is this hardcoded?

Comment (yquansah) [reply plugins/inputs/redis/redis_test.go:20]: Oh just for testing purposes, i am testing the logic around external calls. The struct needs this method to adhere to the interface

Comment (ssoroka) [reply plugins/inputs/redis/redis_test.go:20]: ah, it's just a test. never mind.

**Review by ssoroka — APPROVED** [review]

**Pull request merged**